pkg/genericapiserver/server/options/server_run_options.go

```diff
@@ -66,7 +66,7 @@ type ServerRunOptions struct {
 func NewServerRunOptions() *ServerRunOptions {
 	return &ServerRunOptions{
 		AdmissionControl:            "AlwaysAdmit",
-		DefaultStorageMediaType:     "application/json",
```

Comment:
Yes, I think kube-apiserver and kube-federation and probably kube-adm all need to explicitly override the default from here.  JSON continues to work ootb with any server, but protobuf only works if there's the extra steps taken. @sttts opinion?